src/varset.jl

```diff
@@ -7,39 +7,38 @@ macro varset(name::Symbol, table)
 
     names = Any[]
     vars = Any[]
+    depwarns = Any[]
     parsed_vars = Any[]
     parameters = Any[]
     parameters_expr = Expr(:parameters)
     inherit_parameters = Any[]
     inherit_parameters_expr = Expr(:parameters)
 
     for row in table
-        if isa(row, Expr) && row.head == :line
-            continue
-        end
-
-        if isa(row, Expr) && isa(row.args[1], Expr)
-            docstr = row.args[2]
-            row = row.args[3]
-        else
-            docstr = ""
-        end
+        isa(row, Expr) && row.head == :line && continue
 
+        hasdocstr = false
         if isa(row, Symbol)
             var = row
             typ = :Any
             default = :nothing
-        elseif row.head == :tuple
-            @assert 2 <= length(row.args) <= 3
-            var = row.args[1]
-            typ = row.args[2]
-            default = length(row.args) > 2 ? row.args[3] : :nothing
+            depwarnstr = ""
+        elseif row.head == :tuple && 2 <= length(row.args) <= 5
+            if isa(row.args[1], String)
+                docstr = row.args[1]
+                hasdocstr = true
+            end
+            var = row.args[hasdocstr+1]
+            typ = row.args[hasdocstr+2]
+            default = length(row.args)-hasdocstr > 2 ? row.args[hasdocstr+3] : :nothing
+            depwarnstr = length(row.args)-hasdocstr > 3 ? row.args[hasdocstr+4] : ""
```

Comment:
So the only thing that determines whether something is deprecated or not is whether it has another docstring line following it? I was thinking that you could explicitly use syntax similar to the @deprecated macro (like explicitly have @deprecated in the first position of the line with the alt text) and then to do some processing to determine whether it's present. It would be nice to be super clear that something is deprecated when looking at the source.

Reply:
i've added docs to the top of src/varset.jl to explain the format of the table.